lib/chopperbot/split.ex

```diff
@@ -112,26 +112,51 @@ defmodule Chopperbot.Split do
   ## Examples
       iex> apply_options([{"a", 300}, {"b", 400}], ["+s"])
       [{"a", 330.0}, {"b", 440.0}]
+      iex> apply_options([{"a", 100}, {"b", 200}], ["-20.5%"])
+      [{"a", 79.5}, {"b", 159.0}]
   """
   @spec apply_options(orders(), options()) :: orders()
-  def apply_options(orders, ["+service" | rest]), do: apply_options(orders, ["+s" | rest])
-  def apply_options(orders, ["+vat" | rest]), do: apply_options(orders, ["+v" | rest])
+  def apply_options(orders, options) do
+    Enum.reduce(options, orders, fn option, modified_orders ->
+      apply_option(modified_orders, option)
+    end)
+  end
+
+  defp apply_option(orders, option) do
+    Enum.map(orders, fn {name, amount} ->
+      new_amount =
+        option
+        |> get_multiplier_from_option()
+        |> Kernel.*(amount)
+        |> rounding_floating_problem()
+
+      {name, new_amount}
+    end)
+  end
```

Comment:
Would this approach make it easier to read instead of reduce?

```ex
  defp apply_options(orders, [option | rest_options]) do
    Enum.map(orders, fn {name, amount} ->
      new_amount =
        option
        |> get_multiplier_from_option()
        |> Kernel.*(amount)
        |> rounding_floating_problem()

      {name, new_amount}
    end)
    |> apply_options(rest_options)
  end
```

Reply:
Looks good, thx 👍